ethereumj-core/src/main/java/org/ethereum/casper/core/CasperTransactionExecutor.java

```diff
@@ -0,0 +1,145 @@
+    }
+
+    public CasperTransactionExecutor(Transaction tx, byte[] coinbase, Repository track, BlockStore blockStore,
+                                     ProgramInvokeFactory programInvokeFactory, Block currentBlock,
+                                     EthereumListener listener, long gasUsedInTheBlock) {
+
+        super(tx, coinbase, track, blockStore, programInvokeFactory, currentBlock, listener, gasUsedInTheBlock);
+    }
+
+    @Override
+    public void init() {
+        super.init();
+        // Already failed on common validation
+        if (execError != null) {
+            return;
+        }
+
+        // More validations for Casper
+
+        // EIP 208
+        if (Arrays.equals(tx.getSender(), Transaction.NULL_SENDER) && toBI(tx.getNonce()).compareTo(BigInteger.ZERO) > 0) {
+            execError(String.format("Null sender transaction should use 0 nonce, %s instead", toBI(tx.getNonce())));
+
+            return;
+        }
+
+        // EIP 208
+        if (Arrays.equals(tx.getSender(), Transaction.NULL_SENDER) &&
+                (toBI(tx.getValue()).compareTo(BigInteger.ZERO) > 0 || toBI(tx.getGasPrice()).compareTo(BigInteger.ZERO) > 0)) {
+            execError(String.format("Null sender transaction should have 0 value (actual %s), " +
+                    "and 0 gasprice (actual: %s)", toBI(tx.getValue()), toBI(tx.getGasPrice())));
+
+            return;
+        }
+    }
+
+    @Override
+    protected boolean isSignatureValid() {
+        return isCasperVote() || super.isSignatureValid();
+    }
+
+    @Override
+    public void execute() {
+
+        if (!readyToExecute) return;
+
+        if (!localCall && !isCasperVote()) {
+            track.increaseNonce(tx.getSender());
+
+            BigInteger txGasLimit = toBI(tx.getGasLimit());
+            BigInteger txGasCost = toBI(tx.getGasPrice()).multiply(txGasLimit);
+            track.addBalance(tx.getSender(), txGasCost.negate());
+
+            if (logger.isInfoEnabled())
+                logger.info("Paying: txGasCost: [{}], gasPrice: [{}], gasLimit: [{}]", txGasCost, toBI(tx.getGasPrice()), txGasLimit);
+        }
+
+        if (tx.isContractCreation()) {
+            create();
+        } else {
+            call();
+        }
+    }
+
+    private boolean isCasperVote() {
+        return isCasperVote(tx, ((CasperProperties) config).getCasperAddress());
+    }
+
+    public static boolean isCasperVote(Transaction transaction, byte[] casperAddress) {
+        if (!Arrays.equals(transaction.getSender(), Transaction.NULL_SENDER))
+            return false;
+        if (casperAddress == null)
+            return false;
+        if (!Arrays.equals(transaction.getReceiveAddress(), casperAddress))
+            return false;
+
+        byte[] dataCopy = new byte[4];
+        System.arraycopy(transaction.getData(), 0, dataCopy, 0, 4);
+        return Arrays.equals(dataCopy, new byte[] {(byte) 0xe9, (byte) 0xdc, 0x06, 0x14});
```

Comment:
Use `FastBytesComparison` to avoid `dataCopy`. Also, let's keep constant array in a static class member.

Reply:
Ok